lib/templates.php

```diff
@@ -17,19 +17,40 @@ function gutenberg_register_template_post_type() {
 	}
 
 	$labels = array(
-		'name' => __( 'Templates', 'gutenberg' ),
+		'name'                  => __( 'Templates', 'gutenberg' ),
+		'singular_name'         => __( 'Template', 'gutenberg' ),
+		'menu_name'             => _x( 'Templates', 'Admin Menu text', 'gutenberg' ),
+		'add_new'               => _x( 'Add New', 'Template', 'gutenberg' ),
+		'add_new_item'          => __( 'Add New Template', 'gutenberg' ),
+		'new_item'              => __( 'New Template', 'gutenberg' ),
+		'edit_item'             => __( 'Edit Template', 'gutenberg' ),
+		'view_item'             => __( 'View Template', 'gutenberg' ),
+		'all_items'             => __( 'All Templates', 'gutenberg' ),
+		'search_items'          => __( 'Search Templates', 'gutenberg' ),
+		'parent_item_colon'     => __( 'Parent Template:', 'gutenberg' ),
+		'not_found'             => __( 'No templates found.', 'gutenberg' ),
+		'not_found_in_trash'    => __( 'No templates found in Trash.', 'gutenberg' ),
+		'archives'              => __( 'Template archives', 'gutenberg' ),
+		'insert_into_item'      => __( 'Insert into template', 'gutenberg' ),
+		'uploaded_to_this_item' => __( 'Uploaded to this template', 'gutenberg' ),
+		'filter_items_list'     => __( 'Filter templates list', 'gutenberg' ),
+		'items_list_navigation' => __( 'Templates list navigation', 'gutenberg' ),
+		'items_list'            => __( 'Templates list', 'gutenberg' ),
 	);
 
 	$args = array(
-		'labels'          => $labels,
-		'description'     => __( 'Templates to include in your theme.', 'gutenberg' ),
-		'public'          => false,
-		'has_archive'     => false,
-		'show_in_rest'    => true,
-		'rest_base'       => 'templates',
-		'capability_type' => array( 'template', 'templates' ),
-		'map_meta_cap'    => true,
-		'supports'        => array(
+		'labels'            => $labels,
+		'description'       => __( 'Templates to include in your theme.', 'gutenberg' ),
+		'public'            => false,
```

Comment:
I tried creating a new template with the title “The About Us Page Template” and upon publishing it got the expected slug of `the-about-us-page-template`:

> ![image](https://user-images.githubusercontent.com/134745/65785911-54caa500-e10a-11e9-91f0-56e815cc292b.png)

However, I am not able to change the slug to be something else:

> ![image](https://user-images.githubusercontent.com/134745/65785971-7d529f00-e10a-11e9-9784-280cd28d609f.png)

Similarly I can't edit the slug in the post itself:

> ![image](https://user-images.githubusercontent.com/134745/65786154-e0443600-e10a-11e9-865d-8e02dde8987d.png)

I presume this is because the post type is not viewable due to it not having `'public' => false` here.

But being able to change the slug is important, especially in the case of non-English installs. 

In addition to template names in the hierarchy, consider also templates intended to override specific posts/pages. For example, I could create an “About” page that has the slug of `about`. Then I could create a special page template for this about page, and the template hierarchy could apply it automatically if the `wp_template` has the slug of `page-about.php`. However, if I create the `wp_template` post with the title of “About Page Template” then it will get the slug of `about-page-template` and it won't automatically apply in the template hierarchy. There would then have to be a way to manually select this template from the About page itself (which should still be a capability).

Related: https://github.com/WordPress/gutenberg/issues/11277 and https://core.trac.wordpress.org/changeset/43855

Reply:
Good point. At least the editing of the slug should be possible from this UI.

Regarding page templates, a few ideas:
* Since we rely on the slugs for all the regular templates, we could align on a specific pattern, e.g. a `wp_template` post starting with `template-` is considered a page template.
* Alternatively we could just allow _any_ template with an arbitrary slug to become a page template by adding a specific meta flag. I think this would cater better for any post types, e.g. it could be a meta key `is_template_for_post_type`, and then the value would be a post type. What do you think?

I think for this PR itself, we should only focus on getting in the minimal UI required to edit templates in a sufficient way.

Reply:
PHP templates allow slugs like  `page-about.php` which could also have `Template Name` metadata to allow it to be selected as a page template via dropdown. So to me it seems the same could be allowed here. So yes, adding some meta to specify what kind of post type a given template is for seems to make sense.

Reply:
>PHP templates allow slugs like page-about.php which could also have Template Name metadata

Yeah, let's keep the current template hierarchy slugs, but support different titles for clarity.